text/0230-telemetry-viewer.md

```diff
@@ -0,0 +1,129 @@
+
+## Internal details
+
+Broadly, the implementation of this viewer should be a collector extension that
+exposes a simple web portal for viewing data along with some sort of data store
+to hold the data emitted. This extension could be bundled with specific
+collector releases, or brought in via the collector builder.
+
+There should be a few options for distribution of the extension -
+
+1. A 'default' collector distribution from the project that includes a basic
+   collector configuration and the viewer extension.
+2. Using the collector builder to create a custom image that includes this
+   extension as well as other custom components.
+3. A standalone binary that can be installed on a local machine or in a
+   codespace or other development environment.
+
+These are not an exhaustive list of deployment options, and I posit that the
+community will create other strategies as well.
+
+## Trade-offs and mitigations
+
+There are two major trade-offs this makes in terms of the ecosystem; One, it
+brings this component into the purview of the OpenTelemetry organization rather
+than leaving it entirely to both independent or commercial development efforts.
+Two, it could seem to presage a more opinionated approach to the collector as a
+product offering rather than as a component or piece of middleware.
+
+To the former point, I believe that the project has a responsibility to define
+requirements and signposts for tooling that we believe would be useful in order
+to not only grow our developer community, but also our contributor community.
+
+To the latter, I would argue that if we seek to become a 'native instrumentation
+layer' for cloud-native systems, it is incumbent upon us to be opinionated about
+how OpenTelemetry should be used, and to provide the development community with
+tooling that makes their lives easier.
+
+## Prior art and alternatives
+
+The biggest piece of prior art or alternative implementation here is the
+existence of open source observability and monitoring projects such as
+Prometheus, Grafana, etc. The counter-argument against this proposal is that by
+creating a secondary toolchain for developers, we would not be skilling them up
+in popular existing tools, and these existing tools satisfy the requirements of
+this proposal already.
+
+To this, I submit the following points:
+
+1. While Prometheus and Jaeger do provide powerful analysis tools and can both
+   be run fairly easily in a local environment, they are not necessarily built
+   for the purpose of this proposal as-is.
+2. There is no CNCF tooling available for Dashboard or Log Storage/Querying;
+   Currently, Grafana and OpenSearch seem to be the most popular and open
+   source-iest options here.
+3. There is a difference in effort requireed for a developer to add a local
+   observability stack vs. swapping out a collector binary, and it's a pretty
+   significant difference.
+
```

Comment:
The existing tools are optimized towards usage by DevOps for monitoring production systems. They are built around defining dashboards, alerts and querying storage, rather than a live view of what is being emitted by the applications. A dev focused tool would concentrate on showing what is being sent over the wire, rather than how I want to monitor the application when it's under load.